src/arch/src/x86_64/interrupts.rs

```diff
@@ -9,11 +9,13 @@ use kvm_bindings::kvm_lapic_state;
 use kvm_ioctls::VcpuFd;
 use utils::byte_order;
 /// Errors thrown while configuring the LAPIC.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error, PartialEq)]
```

Comment:
... and in other Error types

Reply:
> Do people tend to break semver? I would assume that if 1 does not become 2 we should be fine.

Yes, all the time. Since their is no popular commonly used mechanism for checking even big crates break it, even when they try to abide it.

But really I don't mind going for `="1"` if you think its best?

Reply:
`kvm_ioctls::Error` doesn't implement `std::cmp::Eq` unfortunately (I've also merged PR fixing this, but this will require new version publish).

Reply:
why do we need it in the first place? Does thiserror require it?